hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -74,6 +96,130 @@ private boolean hasCryptoAdminKey(final ContractUpdateTransactionBody op) {
 
     @Override
     public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+        final var txn = requireNonNull(context).body();
+        final var op = txn.contractUpdateInstanceOrThrow();
+        final var target = op.contractIDOrThrow();
+
+        final var accountStore = context.readableStore(ReadableAccountStore.class);
+        final var toBeUpdated = accountStore.getContractById(target);
+        validateSemantics(toBeUpdated, context, op);
+        final var changed = update(toBeUpdated, context, op);
+
+        context.serviceApi(TokenServiceApi.class).updateContract(changed);
+    }
+
+    public Account update(
+            @NonNull final Account contract,
+            @NonNull final HandleContext context,
+            @NonNull final ContractUpdateTransactionBody op) {
+        final var builder = contract.copyBuilder();
+        if (op.hasAdminKey()) {
+            if (EMPTY_KEY_LIST.equals(op.adminKey())) {
+                builder.key(contract.key());
+            } else {
+                builder.key(op.adminKey());
+            }
+        }
+        if (op.hasExpirationTime()) {
+            if (contract.expiredAndPendingRemoval()) {
+                builder.expiredAndPendingRemoval(false);
+            }
+            builder.expirationSecond(op.expirationTime().seconds());
+        }
+        if (op.hasAutoRenewPeriod()) {
+            builder.autoRenewSeconds(op.autoRenewPeriod().seconds());
+        }
+        if (affectsMemo(op)) {
+            final var newMemo = op.hasMemoWrapper() ? op.memoWrapper() : op.memo();
+            context.attributeValidator().validateMemo(newMemo);
+            builder.memo(newMemo);
+        }
+        if (op.hasStakedAccountId()) {
+            if (SENTINEL_ACCOUNT_ID.equals(op.stakedAccountId())) {
+                builder.stakedAccountId((AccountID) null);
+            } else {
+                builder.stakedAccountId(op.stakedAccountId());
+            }
+        } else if (op.hasStakedNodeId()) {
+            builder.stakedNodeId(op.stakedNodeId());
+        }
+        if (op.hasDeclineReward()) {
+            builder.declineReward(op.declineReward());
+        }
+        if (op.hasAutoRenewAccountId()) {
+            builder.autoRenewAccountId(op.autoRenewAccountId());
+        }
+        if (op.hasMaxAutomaticTokenAssociations()) {
+            final var ledgerConfig = context.configuration().getConfigData(LedgerConfig.class);
```

Comment:
⚠️ We appear to be missing [this check](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractUpdateTransitionLogic.java#L188) from mono-service that fails with `NOT_SUPPORTED` if any `ContractUpdate` tries to configure auto-associations.